src/coreservices.cpp

```diff
@@ -102,7 +102,8 @@ inline QLocale inputLocale() {
 namespace mixxx {
 
 CoreServices::CoreServices(const CmdlineArgs& args)
-        : m_runtime_timer(QLatin1String("CoreServices::runtime")),
+        : m_pLV2Backend(nullptr),
```

Comment:
We should initialize this plain pointer in the header with `LV2Backend* m_pLV2Backend{};`

Reply:
EffectsBackends don't even belong here. This is fixed in #2618. DlgPrefLV2 has been removed.

Reply:
Please do not touch any effects code before merging #2618. I refuse to fix any merge conflicts from others doing so.

Reply:
> We should initialize this plain pointer in the header with LV2Backend* m_pLV2Backend{};

In Mixxx we only header initialization with exceptions because this introduces a hidden static constructor.
https://github.com/mixxxdj/mixxx/wiki/Coding-Guidelines#non-static-data-member-initializers

Reply:
I strongly disagree. It doesn't make any sense to repeat all plain pointer members and initialize them explicitly with `nullptr`. This introduces useless noise. It's much more safe to initialize them directly, especially if you have many of them, see WTrackMenu. By using managed pointers this issue will hopefully disappear with time.

Reply:
We should revise the coding guidelines. Neither do I understand the motivation nor the issue that should be solved.

Reply:
"...is used if the member is omitted from the member initializer list of a constructor".

What's wrong with providing a sensible default value to prevent uninitialized members???

Reply:
This whole discussion about what to do with the pointer is moot. It is removed from this code already in #2618.

Reply:
> What's wrong with providing a sensible default value to prevent uninitialized members???

Nothing. 

The question is just where it should be done. 
The normal or lets say legacy place is the initialization list in the constructor.
If we keep this habit we have a single place to look at. 

Here we have the case where some variables are const initialized and some from parameter values. So you need to look at two places. The header initialization causes an extra static inline constructor and forces recompile more files if we change that value.    

The solution in WTrackMenu initializes many pointers twice. One time the stat inline constructor and one time in the real constructor. This is IMHO even more an issue, because the reader might think that he has to deal with a null pointer when he looks into the header which is not the case.  

I am not opposing to adjust our policy if we have sensible new ideas. But for now I like to keep the current solution here.

Reply:
I suggest to choose the header-only modification for minimizing the amount of changes and potential merge conflicts. In the end it doesn't matter, so let's be pragmatic.